prototype.py

```
@@ -169,16 +172,150 @@ def evaluate_strategy(self, strategy: Callable[[int, int], bool], duration_secon
+                power=params["new_power"],
+                modulus_filters=parent.modulus_filters[:],
+                smoothness_bound=parent.smoothness_bound,
+                min_small_prime_hits=parent.min_small_prime_hits,
+            )
+
+        elif mutation_type == "add_filter":
+            new_filters = parent.modulus_filters[:]
+            if len(new_filters) < 4:  # 最大4フィルタまで
+                new_filters.append((params["modulus"], params["residues"]))
+            else:
+                logger.warning(f"Cannot add filter: maximum limit (4) reached, keeping parent strategy")
```

Ruff (0.14.1)

f-string without any placeholders

Remove extraneous `f` prefix

(F541)